### PR TITLE
Add an option to export data in a format compatible with Kuzzle fixtures

### LINKED ISSUE #56: Use same format for dump files as for mappings, securities, and fixtures
We have too many formats: Kourou, Kuzzle and the bulk format.

We should use Kuzzle format for mappings, users, profiles and roles. Kourou can try to guess which file correspond to which kind of data (frequency analysis, pattern recognition)

For fixtures, if there is less than 2000 documents (for example), then we can use the standard fixtures format, otherwise Kourou will use JSONL format.

### PR DESCRIPTION
**Pull request by jenow**
## What does this PR do?

Add option to export in JSON understandable for Kuzzle fixtures.

fix #56 

### How should this be manually tested?

```
kourou index:export my-index --format=raw
kourou collection:export my-index my-collection --format=raw
```

### CODE REVIEW

**Review by Aschen — CHANGES_REQUESTED** [review]
It could be interesting to have a better description of the commands, particularly to explain why choosing one or the other format (use with Kourou vs use as Kuzzle fixtures)

File: src/commands/collection/export.ts [hunk 1]
```diff
@@ -25,6 +25,10 @@ export default class CollectionExport extends Kommand {
     editor: flags.boolean({
       description: 'Open an editor (EDITOR env variable) to edit the query before sending'
     }),
+    format: flags.string({
+      description: 'JSONL or kuzzle - kuzzle will export in Kuzzle format',
```
Comment (Aschen) [inline src/commands/collection/export.ts:29]:
```suggestion
      description: '"jsonl" or "kuzzle" - "kuzzle" will export in Kuzzle format',
```

File: src/commands/index/export.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { kuzzleFlags } from '../../support/kuzzle'
 import { dumpCollectionData, dumpCollectionMappings } from '../../support/dump-collection'
 
 export default class IndexExport extends Kommand {
-  static description = 'Exports an index (JSONL format)'
+  static description = 'Exports an index (JSONL format or kuzzle Kuzzle format)'
```
Comment (Aschen) [inline src/commands/index/export.ts:11]:
```suggestion
  static description = 'Exports an index (JSONL or Kuzzle format)'
```

File: src/commands/index/export.ts [hunk 3]
```diff
@@ -19,6 +19,10 @@ export default class IndexExport extends Kommand {
       description: 'Maximum batch size (see limits.documentsFetchCount config)',
       default: '2000'
     }),
+    format: flags.string({
+      description: 'JSONL or kuzzle - kuzzle will export in Kuzzle format',
```
Comment (Aschen) [inline src/commands/index/export.ts:23]:
```suggestion
      description: '"jsonl" or "kuzzle" - "jsonl" will export in Kourou format and "kuzzle" will export in Kuzzle fixtures format',
```

File: src/commands/index/export.ts [hunk 4]
```diff
@@ -19,6 +19,10 @@ export default class IndexExport extends Kommand {
...
+      default: 'JSONL'
```
Comment (Aschen) [inline src/commands/index/export.ts:24]:
```suggestion
      default: 'jsonl'
```

File: src/support/dump-collection.ts [hunk 5]
```diff
@@ -6,9 +6,9 @@ import cli from 'cli-ux'
 // tslint:disable-next-line
 const ndjson = require('ndjson')
 
-export async function dumpCollectionData(sdk: any, index: string, collection: string, batchSize: number, destPath: string, query: any = {}) {
+export async function dumpCollectionData(sdk: any, index: string, collection: string, batchSize: number, destPath: string, query: any = {}, format = 'JSONL') {
```
Comment (Aschen) [inline src/support/dump-collection.ts:9]:
```suggestion
export async function dumpCollectionData(sdk: any, index: string, collection: string, batchSize: number, destPath: string, query: any = {}, format = 'jsonl') {
```

File: src/support/dump-collection.ts [hunk 6]
```diff
@@ -6,9 +6,9 @@ import cli from 'cli-ux'
...
   const collectionDir = path.join(destPath, collection)
-  const filename = path.join(collectionDir, 'documents.jsonl')
+  const filename = path.join(collectionDir, (format.toUpperCase() === 'JSONL' ? 'documents.jsonl' : 'documents.json'))
```
Comment (Aschen) [inline src/support/dump-collection.ts:11]:
```suggestion
  const filename = path.join(collectionDir, (format.toLowerCase() === 'jsonl ? 'documents.jsonl' : 'documents.json'))
```

File: src/support/dump-collection.ts [hunk 7]
```diff
@@ -36,7 +36,9 @@ export async function dumpCollectionData(sdk: any, index: string, collection: st
 
   fs.mkdirSync(collectionDir, { recursive: true })
 
-  await writeLine({ type: 'collection', index, collection })
+  if (format.toUpperCase() === 'JSONL') {
```
Comment (Aschen) [inline src/support/dump-collection.ts:39]:
```suggestion
  if (format.toLowerCase() === 'jsonl') {
```

File: src/support/dump-collection.ts [hunk 8]
```diff
@@ -45,27 +47,47 @@ export async function dumpCollectionData(sdk: any, index: string, collection: st
...
+
   progressBar.stop()
   ndjsonStream.end()
   writeStream.end()
 
   return waitWrite
 }
 
-export async function dumpCollectionMappings(sdk: any, index: string, collection: string, destPath: string) {
+export async function dumpCollectionMappings(sdk: any, index: string, collection: string, destPath: string, format = 'JSONL') {
```
Comment (Aschen) [inline src/support/dump-collection.ts:90]:
```suggestion
export async function dumpCollectionMappings(sdk: any, index: string, collection: string, destPath: string, format = 'jsonl') {
```

File: src/support/dump-collection.ts [hunk 9]
```diff
@@ -82,5 +104,5 @@ export async function dumpCollectionMappings(sdk: any, index: string, collection
     }
   }
 
-  fs.writeFileSync(filename, JSON.stringify(dump, null, 2))
+  fs.writeFileSync(filename, JSON.stringify((format.toUpperCase() === 'JSONL' ? dump : mappings), null, 2))
```
Comment (Aschen) [inline src/support/dump-collection.ts:107]:
```suggestion
  fs.writeFileSync(filename, JSON.stringify((format.toLowerCase() === 'jsonl ? dump : mappings), null, 2))
```

**Review by Aschen — CHANGES_REQUESTED** [review]

File: src/commands/index/export.ts [hunk 10]
```diff
@@ -20,7 +20,7 @@ export default class IndexExport extends Kommand {
       default: '2000'
     }),
     format: flags.string({
-      description: '"jsonl" or "kuzzle" - "jsonl" will export in Kourou format and "kuzzle" will export in Kuzzle fixtures format',
+      description: '"jsonl or kuzzle - kuzzle will export in Kuzzle format usable for internal fixtures and jsonl will be usable to re import those data whith kourou',
```
Comment (Aschen) [inline src/commands/index/export.ts:23]:
```suggestion
      description: '"jsonl or kuzzle - kuzzle will export in Kuzzle format usable for internal fixtures and jsonl will be usable to re import those data with kourou',
```

**Review by Aschen — APPROVED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

File: README.md [hunk 11]
```diff
@@ -355,14 +355,26 @@ OPTIONS
   --as=as                  Impersonate a user
   --batch-size=batch-size  [default: 2000] Maximum batch size (see limits.documentsFetchCount config)
   --editor                 Open an editor (EDITOR env variable) to edit the query before sending
+
+  --format=format          [default: JSONL] "jsonl or kuzzle - kuzzle will export in Kuzzle format usable for internal
+                           fixtures and jsonl will be usable to re import those data whith kourou
```
Comment (scottinet) [inline README.md:360]:
```suggestion
                           fixtures and jsonl allows to import that data back with kourou
```

File: README.md [hunk 12]
```diff
@@ -722,13 +734,24 @@ OPTIONS
   --api-key=api-key        Kuzzle user api-key
   --as=as                  Impersonate a user
   --batch-size=batch-size  [default: 2000] Maximum batch size (see limits.documentsFetchCount config)
+
+  --format=format          [default: jsonl] "jsonl or kuzzle - kuzzle will export in Kuzzle format usable for internal
+                           fixtures and jsonl will be usable to re import those data whith kourou
```
Comment (scottinet) [inline README.md:739]:
```suggestion
                           fixtures and jsonl allows to import that data back with kourou
```

File: src/commands/collection/export.ts [hunk 13]
```diff
@@ -25,6 +25,10 @@ export default class CollectionExport extends Kommand {
     editor: flags.boolean({
       description: 'Open an editor (EDITOR env variable) to edit the query before sending'
     }),
+    format: flags.string({
+      description: '"jsonl or kuzzle - kuzzle will export in Kuzzle format usable for internal fixtures and jsonl will be usable to re import those data whith kourou',
```
Comment (scottinet) [inline src/commands/collection/export.ts:29]:
```suggestion
      description: '"jsonl or kuzzle - kuzzle will export in Kuzzle format usable for internal fixtures and jsonl allows to import that data back with kourou',
```

File: src/commands/index/export.ts [hunk 14]
```diff
@@ -19,6 +19,10 @@ export default class IndexExport extends Kommand {
       description: 'Maximum batch size (see limits.documentsFetchCount config)',
       default: '2000'
     }),
+    format: flags.string({
+      description: '"jsonl or kuzzle - kuzzle will export in Kuzzle format usable for internal fixtures and jsonl will be usable to re import those data with kourou',
```
Comment (scottinet) [inline src/commands/index/export.ts:23]:
```suggestion
      description: '"jsonl or kuzzle - kuzzle will export in Kuzzle format usable for internal fixtures and jsonl will be usable to import that data back with kourou',
```

**Review by scottinet — CHANGES_REQUESTED** [review]

**Review by scottinet — APPROVED** [review]

**Pull request merged**